Feature/style emotioncard

Leon910 opened the pull request.

JessicaLoers commented on components/BookmarkButton.js:
```diff
@@ -11,8 +11,8 @@ export default function BookmarkButton({ isBookmarked, onToggleBookmark }) {
   position: absolute;
-  top: 10px;
-  right: 10px;
+  top: 1px;
```
The use of `top: 1px` may seem unusual and could create slight alignment issues.

```suggestion
  top: 0;
```

_Little UI notes_ ✏️

Even though it is not part of this PR and more of a QA issue, I would like to briefly mention... 

- That hover effects not available on mobile devices and the active state (scaling down and adding a shadow) can also appear glitchy, as it may look like a bug.
- Although  there’s no fixed size for a toggle (bookmark) button, but there are established guidelines for interactive elements. 
  -  For mobile, Google’s Material Design recommends a minimum touch [target size](https://m3.material.io/components/icon-buttons/specs#2b6575b0-cfa3-494e-9509-d7c1949d11f2) of 48x48 px
  - For desktop, buttons can be smaller since mouse clicks allow for more precision. Typically, a 32-40 px height is a good baseline. 
  - The bookmark button should be large enough for easy access, especially if it’s a frequently used feature.

JessicaLoers commented on components/EmotionCard.js:
```diff
@@ -52,15 +64,39 @@ const StyledLink = styled(Link)`
-  align-items: left;
-  font-size: 1rem;
+const StyledOuterBox = styled.div`
+  background-color: #e0e1f0;
+  border: 1px solid #d3d3d3;
+  border-radius: 0.5rem;
+  box-shadow: 0px 1px 4px #00000029;
```
Just a suggestion: Use custom variables for colors and shadows to make future changes easier.

JessicaLoers commented on components/EmotionCardContent.js:
```diff
@@ -1,25 +1,73 @@
+  align-items: center;
+  padding: 5px;
+
+  @media (max-width: 374px) {
+    grid-template-columns: 1fr;
+    grid-template-rows: auto auto auto auto;
+    grid-template-areas:
+      "emoji"
+      "emotionType"
+      "notes"
+      "intensity";
+  }
+`;
```
Please review this media query: Do you intend for the content to stack vertically on smaller devices?

Currently, this layout changes to a single-column format for screens under 480px width 🤔

JessicaLoers commented on components/EmotionCardContent.js:
```diff
@@ -1,25 +1,73 @@
-  width: 40px;
-  height: 40px;
+  grid-area: emoji;
+  align-self: self-start;
```
Fix Typo: `self-start` is incorrect; `start` is the correct value.

```suggestion
  align-self: start;
```

JessicaLoers commented on components/EmotionCardContent.js:
```diff
@@ -1,25 +1,73 @@
+  grid-area: emotionType;
+  margin: 0;
+  padding-left: 28px;
+  font-weight: bold 600;
```
Fix Typo: Separate `bold` from the numeric font weight
```suggestion
  font-weight: 600;
```

JessicaLoers commented on components/EmotionCardContent.js:
```diff
@@ -1,25 +1,73 @@
+  align-self: start;
+  margin: 6px 12px;
+  text-align: right;
+  font-weight: bold 600;
```
```suggestion
  font-weight: 600;
```

JessicaLoers commented on components/EmotionForm.js:
```diff
@@ -138,7 +138,7 @@ const StyledFormContainer = styled.div`
   border: 1px solid #d3d3d3;
   border-radius: 0.5rem;
-  box-shadow: 0 2px 4px #000;
+  box-shadow: 0px 1px 4px #00000029;
```
As mentioned, using custom variables for colors and shadows would save you time and effort  and create a more consistent design.  😻


For example, you could define `--shadow ` and reference them here instead of hardcoding values.

JessicaLoers reviewed (commented):
**Hey Team**

I think the Emotion Card redesign is very well done 💙 – the layout and structure are clear and intuitive. 
 
However, the PR isn’t approved yet, as I have one remaining question about the media query: please double-check if the current setup achieves the intended behavior on smaller devices. 
**If this is exactly how you’d like it to work, just let me know, and I’ll switch the gray check to a green one.** 🫡

Jokes aside, I also strongly recommend using custom variables for colors and shadows. This will save you a lot of work and help maintain design consistency. As the app grows, adjustments will only increase, and centralising these styles will prevent inconsistencies –  it would be a shame if your great design lost any of its cohesion!

Leon910 commented:
Hey @JessicaLoers ! Thank you a lot for your great suggestions and useful hints and code suggestions. 

We applied most of them as intended. The @media I implemented because I thought if there are screens with a width lower than iPhone SE (375px) it might be helpful but since we do not assume a smaller screen I removed it again.

ahohnsen commented on components/EmotionCard.js:
```diff
@@ -52,15 +64,39 @@ const StyledLink = styled(Link)`
+  background-color: var(--color-background);
+  border: 1px solid var(--color-border);
+  border-radius: 0.5rem;
+  box-shadow: 0px 1px 4px var(--color-shadow);
```
If the value is 0, you can omit the unit.

```suggestion
  box-shadow: 0 1px 4px var(--color-shadow);
```

ahohnsen commented on components/EmotionForm.js:
```diff
@@ -136,9 +136,9 @@ const StyledFormContainer = styled.div`
+  border: 1px solid (--color-border);
   border-radius: 0.5rem;
-  box-shadow: 0 2px 4px #000;
+  box-shadow: 0px 1px 4px var(--color-shadow);
```
Same as above.

```suggestion
  box-shadow: 0 1px 4px var(--color-shadow);
```

ahohnsen reviewed (approved):
Well done! 🚀 

I left a few tiny comments. 
In addition to color variables, you could also store other recurring values as variables, such as common padding or margin values, or even the size of box shadows, etc.
